### PR TITLE
Redis enterprise .net SDK for new API version 2024-09-01-preview

### PR DESCRIPTION
**Pull request by priyankarking**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/CHANGELOG.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
 # Release History
 
-## 1.2.0-beta.1 (Unreleased)
+## 1.2.0-beta.1 (2024-09-01-preview)
```
Comment (archerzz) [inline sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/CHANGELOG.md:3]: the content in the paratheses is the target release date for the SDK, see: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges for details.

**Review by archerzz — APPROVED** [review]

File: sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/api/Azure.ResourceManager.RedisEnterprise.netstandard2.0.cs [hunk 2]
```diff
@@ -729,4 +917,23 @@ public RedisPersistenceSettings() { }
         string System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.RedisPersistenceSettings>.GetFormatFromOptions(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
         System.BinaryData System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.RedisPersistenceSettings>.Write(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
     }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct RedundancyMode : System.IEquatable<Azure.ResourceManager.RedisEnterprise.Models.RedundancyMode>
```
Comment (ArthurMa1978) [inline sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/api/Azure.ResourceManager.RedisEnterprise.netstandard2.0.cs:921]: The current name of this struct is too generic. Please rename it to something more descriptive.

File: sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/api/Azure.ResourceManager.RedisEnterprise.netstandard2.0.cs [hunk 3]
```diff
@@ -267,6 +385,24 @@ public ForceUnlinkRedisEnterpriseDatabaseContent(System.Collections.Generic.IEnu
         string System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.ForceUnlinkRedisEnterpriseDatabaseContent>.GetFormatFromOptions(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
         System.BinaryData System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.ForceUnlinkRedisEnterpriseDatabaseContent>.Write(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
     }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct HighAvailability : System.IEquatable<Azure.ResourceManager.RedisEnterprise.Models.HighAvailability>
```
Comment (ArthurMa1978) [inline sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/api/Azure.ResourceManager.RedisEnterprise.netstandard2.0.cs:389]: The current name of this struct is too generic. Please rename it to something more descriptive.

Comment (priyankarking) [reply sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/api/Azure.ResourceManager.RedisEnterprise.netstandard2.0.cs:389]: what will be the effect on changing the name here ?

File: sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/api/Azure.ResourceManager.RedisEnterprise.netstandard2.0.cs (hunk 2 above)
Comment (priyankarking) [reply sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/api/Azure.ResourceManager.RedisEnterprise.netstandard2.0.cs:921]: what will be the effect on changing the name here ?

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**